anima_train.py

```
@@ -129,13 +85,14 @@ def train(self, args):
             "blocks_to_swap is not supported with unsloth_offload_checkpointing"
 
         # Attention: validate availability
-        if getattr(args, 'flash_attn', False):
-            try:
-                import flash_attn  # noqa: F401
-                logger.info("Flash Attention enabled for DiT blocks")
-            except ImportError:
-                logger.warning("flash_attn package not installed, falling back to PyTorch SDPA")
-                args.flash_attn = False
+    if getattr(args, 'flash_attn', False):
```

Ruff (0.15.11)

[error] Undefined name `args`

(F821)